[16.0] [MIG] base_geoengine : migration from odoo 14 to odoo 16

kouffsamuel opened the pull request.

lmignon committed:
[FIX] Reworks docstring

lmignon committed:
[IMP] Use V8 API

lmignon committed:
[FIX] Use isinstance operator in place of the previous hack
Shapely now provide a BaseGeometry class as base class for all Geometry objects

Commit:
[FIX] run tests in V8

Commit:
[FIX] With new API: Non stored fields doesn't have column attribute

lmignon committed:
[ADD] Add acsone in the list of authors

nbessi committed:
Fix MissingError wrong argument number

And improve error message

nbessi committed:
Improve error message

nbessi committed:
Styling

nbessi committed:
Fix styling

nbessi committed:
Style

nbessi committed:
Experiment to remove of history

nbessi committed:
Add type testing when browsing

nbessi committed:
Avoid exception when query result is empty on map view

nbessi committed:
White space cleanup

nbessi committed:
Remove experimental dead code

Commit:
Allows doc to be build on github pages

lmignon committed:
[FIX] tests

lmignon committed:
[FIX] Declare admin user as member of 'Geoengine admin' group

Commit:
Fix travis documentation location

Commit:
Upgrade doc to postgis 2.0

lmignon committed:
[FIX] Add bug tracker url to the module description

Commit:
Fix addons path by loading them and pep8

lmignon committed:
Use the Nominatim service of OpenStreetMap to retrieve OSM data by address

lmignon committed:
[PEP8]

Commit:
Fix autodoc

Commit:
Fix docstring indentation

Commit:
Add pointer to documentation

Commit:
Fix copyright to OCA

Commit:
Renaming to geospatial

lmignon committed:
[IMP] OL3

lasley committed:
* updated Google maps URI to relative scheme
* updated js.mapbox theme URI to relative scheme and their Cloudfront address so that SSL cert is valid

lmignon committed:
[FIX] map styling and fullscreen mode

lmignon committed:
chromojs

lmignon committed:
FIX legend
Implement propotional
Improve alpha channel management (everythong is ready to enable its configuration on the raster layer)

lmignon committed:
[FIX] When testing geoegine view, only keep relevant keys from values returned by fields_view_get

dreispt committed:
Update installation instructions

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

yvaucher committed:
[MIGR] fix menu icons path for base_geoengine

yvaucher committed:
Remove version="7.0" from tests

yvaucher committed:
[MIGR] base_geoengine
- separate view and widgets in 2 files
- change js models using 'odoo.define'

yvaucher committed:
[MIGR] Separate common function for widget and view to create layers in a mixin abstract model

yvaucher committed:
[MIGR][FIX] Don't call order_by if order is not defined

yvaucher committed:
[MIGR] ensure geoengine view is loaded after container is rendered

yvaucher committed:
Not pretending we implement limit and offset

yvaucher committed:
Fix search returning 0 elements on geo view

yvaucher committed:
Patch mapfish to avoid trying to make more classes than element to classify

mdietrichc2c committed:
Improve patch to classify points to make sure there are as many classes as possible

yvaucher committed:
Fix a typo on default key

yvaucher committed:
Fix tab event listener to display map widget in form view in last tabs

yvaucher committed:
[FIX] display translation of field string instead of field name and display values of selection field instead of technical value in info map box in geoengine view

yvaucher committed:
[FIX] base_geoengine - allow to use multiple icons on your maps

The feature to set a symbol with symbol_url or symbol_binary on vector
layers was broken. This replace those fields by a new object vector
symbol on which you can specify which symbol to use depending on a field
value.

yvaucher committed:
Allow to drag pan through polygons

yvaucher committed:
Refactor WMTS to be able to add a generic config

And extract Swisstopo to move it in a new module

yvaucher committed:
Fix srid value read which was unset in field properties

yvaucher committed:
By default use OSM

yvaucher committed:
Add parameters to define extends and projection

yvaucher committed:
Fix path of openerp for doc build

yvaucher committed:
Remove google from selection as it is currently broken

yvaucher committed:
Add geo_contains operator to geo_search method.

geo_contains calls St_Contains SQL search.

Plus fix search to set SRID to avoid mixin srid issue.

yvaucher committed:
Target only the right tab where the map is to add listener.

fredj committed:
Ensure we call add_tab_listener only once

yvaucher committed:
Add a button on map widget to clear geometry from the map

yvaucher committed:
Fix _setup_fields signature due to OCA/OCB@87d160a
Otherwise partial value is interpreted as context and it will fails on
Odoo loading

yvaucher committed:
Bump version

yvaucher committed:
Adapt test to mandatory partial attribute in _setup_fields

yvaucher committed:
Adapt tests

yvaucher committed:
Bump version

yvaucher committed:
Address the right container when drawing the widget map

yvaucher committed:
Fix widget initialized too early in second tab thus having bad size

yvaucher committed:
Fix offset issues on the map by reseting offset when scroll bar is used

yvaucher committed:
Don't zoom when setting or changing a geometry

yvaucher committed:
Add an option to set a default zoom to avoid showing a map that you always need to zoom out to get useful informations

yvaucher committed:
No key argument in js, set an optional argument instead with default value

yvaucher committed:
Add an action on map info box to open the selected object in form view

yvaucher committed:
Update Roadmap

yvaucher committed:
Bump version

yvaucher committed:
Fix z-index of Openlayers map container to ensure it doesn't hide search drop down list

yvaucher committed:
Fix unitialized map on 2nd tab of popup forms

yvaucher committed:
Bump version

yvaucher committed:
Fix geoengine tests with addition of default_zoom

yvaucher committed:
Add selection controlers

yvaucher committed:
Change display of map box info if multiple features are selected

yvaucher committed:
Add button to filter by selected geo features

yvaucher committed:
Implement button to switch in box selection

yvaucher committed:
Improve button to filter features

alan196 committed:
[FIX] from_latlon method srdi was incorrect

damdam-s committed:
bump OL version to 3.18.2

yvaucher committed:
Remove disabling stopdown as it is included in OL3

lmignon committed:
Add click on feature event

yvaucher committed:
Fix Mapinfo box + set cursor pointer

yvaucher committed:
fitExtent is deprecated by fit

yvaucher committed:
Fix css missing bracket

yvaucher committed:
Fix feature counter on view map

yvaucher committed:
Port creation of filter on selection

yvaucher committed:
Improve filter selection button to fix with open single element button

yvaucher committed:
Fix extent error when no feature are present

yvaucher committed:
Set fa icon for geo selection filter

Commit:
Remove MapQuest

see https://github.com/openlayers/ol3/releases/tag/v3.17.0

Commit:
Fix Represenation Mode colored label

yvaucher committed:
Add a selection dragbox when holding SHIFT

yvaucher committed:
[IMP] multi layer selection of records on Geoengine map
count selected records instead of features

Commit:
Fix map size error when open record in a fullscreen map.

damdam-s committed:
geoengine widget first part:
Display the map and geometry plus start point edition

yvaucher committed:
Remove ol alias

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

yvaucher committed:
Define edit layer of geoengine widget

yvaucher committed:
Add modify interaction on geoengine widget (modify interaction needs to be created after map rendering)

yvaucher committed:
Add comment on restrictedExtent

yvaucher committed:
remove comment

yvaucher committed:
Remove google raster layer deprecated option

yvaucher committed:
Remove former ol2 and unused libraries

yvaucher committed:
Remove former fix for scrolling as it is not needed with OL3

yvaucher committed:
Fix onchange for modify control

yvaucher committed:
Fix geostat color legend position

yvaucher committed:
Reimplement button to clear geometry from edition widget

yvaucher committed:
Update license headers

yvaucher committed:
[PORT] base_geoengine to 10.0 - remove old API

yvaucher committed:
jslint

Commit:
[FIX] #132: geo columns were created by odoo base models

mark fields type as manual let geogine the authority to create geom fields

benwillig committed:
[FIX] Column should be created if GeoField is computed and stored

benwillig committed:
[FIX] Cache value for a geo field should be the the hex code. convert_to_record must always return a BaseGeometry instance

benwillig committed:
[ADD] New method to convert 2 points to a LineString

benwillig committed:
[FIX] Fixed tests. Added new tests for new methods

benwillig committed:
[FIX] Pylint

benwillig committed:
[FIX] Handle convert_to_record case when there is no value or if it's not the expected one

benwillig committed:
[FIX] Do not hardcode string during line creation from points

benwillig committed:
[IMP] Allow to define layers which will be active on startup

benwillig committed:
[ADD] Allow to display a configurable popup on map layers.

benwillig committed:
[IMP] Allow to change the size of the drawn circle

benwillig committed:
[IMP] Allow to pass a specific srid to GeoLine.from_points method

dek-odoo committed:
[Add] Allow to display labels on polygons for layers

ecino committed:
[FIX] Replace openerp imports

dek-odoo committed:
Fix Lint & Travis CI issues

dek-odoo committed:
[10, Add] Allow to set opacity on layers (#165)

* Add: Layer opacity

ecino committed:
[FIX] typo in base_geoengine javascript. (#166)

Tonow-c2c committed:
Play python 2to3

Tonow-c2c committed:
Remove coding: utf-8

Tonow-c2c committed:
Readme split in readme folder

yvaucher committed:
Fix signature change on _setup_fields

yvaucher committed:
_select_column_data is replaced by tools.table_columns

commit odoo/d024c7602146ba293c0082303fd476c8613daedc

yvaucher committed:
Change signature of convert_to_column to to introduction of multi values used by Monetary field

yvaucher committed:
Use new MVC approach

asaunier committed:
Fix geoengine view for Odoo 11

asaunier committed:
Fix geoengine widget for Odoo 11

asaunier committed:
Reorganise geoengine files

asaunier committed:
Add base_geoengine README back

Add note about geometry fields to be listed in geoengine view definitions

asaunier committed:
Add support of onchange requests to convert_to_record

asaunier committed:
Update WMTS backgrounds creation

Fix some linting issues

asaunier committed:
Fix JS lint errors

asaunier committed:
Fix travis tests

asaunier committed:
Fix layer switcher

asaunier committed:
Improve handling of maps in tabs

asaunier committed:
Improve WMTS background layers creation

asaunier committed:
Upgrade to OL 3.20.1

Commit:
[IMP] Add custom type support for colored render

asaunier committed:
Move base_geoengine tests to a dedicated module

It prevents closed cursor errors, most likely dued to using a mock in the tests.

yvaucher committed:
Mig base_geoengine to version 12.0

yvaucher committed:
Add geo fields to odoo.fields

This allow to declare a Geo field using fields instead of having
to import them directly from geoengine module.

yvaucher committed:
Use intheritance of 'base' for GeoModel

With this import of GeoModel is not needed anymore giving to all classes the ability to have geo fields.

yvaucher committed:
Avoid creation of columns on abstract models

Using geo fields on AbstractModels is made possible.

yvaucher committed:
pyeval was renamed pyUtils

yvaucher committed:
Move base_geoengine readme in the right folder

OCA-git-bot committed:
[UPD] README.rst

yvaucher committed:
Add USAGE in base_geoengine

OCA-git-bot committed:
[UPD] README.rst

yvaucher committed:
Make MissingError translatable for auto install of postgis extension, plus fix some typos

yvaucher committed:
Simplify super calls with python 3 syntax

fredj committed:
Use EPSG:3857 instead of EPSG:900913

EPSG:3857 is the official name for the spherical pseudo-mercator coordinate system.
The unofficial 900913 code is deprecated and should not be used as the default anymore.

fredj committed:
Remove server_resolutions field in WMTS options

Deprecated in OpenLayers >3

asaunier committed:
[12.0][ADD] Support of Swisstopo layers

yvaucher committed:
JS lint

- ESlint - ignore libs imported as global

yvaucher committed:
Update OL version to OL 4.6.5

Change fit(extent) as map size is not necessary anymore.
And improve the test on extent, to not try to fit extent when
no features are defined.

yvaucher committed:
Fix dimension field label

yvaucher committed:
Give description to geoengine view models

yvaucher committed:
Fix duplicate title in README.rst

yvaucher committed:
Remove double title in README.rst - in source too this time

yvaucher committed:
Fix selection of ttype field on ir.model.fields

Replace the method by a the use of `selection_add` in order
to allow other modules to extend the list of field type.

Otherwise, it would crash on when installing for instance
both base_geoengine and base_sparse_field.

yvaucher committed:
Bump version

yvaucher committed:
Fix state of field for geo field to 'base'

It was previously state `manual`
Usage of `manual` is for custom fields created from the interface such
as fields prefixed by `x_`.
In commit odoo/odoo@547118d manual field don't have an xmlid anymore.

Thus it had the side effect to fail when loading definition of vector layers
from an XML file as an XMLID is required to reference the field.
The XMLID would not be found.

Fixes #206

yvaucher committed:
Fix geometry field creation missing configurations

After changing the field in state `base` instead of `manual`.
Those fields were created without configuration.
srid would be 0 gemetry type in geometry_columns table would be
GEOMETRY instead of POINT, LINESTRING, ...

This fixes the creation of geometry to use AddGeometryColumn again.

Commit:
[13.0][MIG] base_geoengine

yvaucher committed:
Update CONTRIBUTORS

A lot of names were missing and some names were most probably added by error.

yvaucher committed:
Fix Selection field display value in map info_box

fixes #198

asaunier committed:
[12.0][IMP] Use default zoom when creating a map with no geometry

grindtildeath committed:
FIX: Refresh geoengine widget after changing record through pager action

alpham-andela committed:
[fix] displaying relation fields in map info box - fixes #224

grindtildeath committed:
FIX: Call render instead of renderMap to ensure geometries layers are loaded

Commit:
run pre-commit

Commit:
remove executable permission from geoenigine_record.js

Commit:
pre-commit with correct config

Commit:
pre-commit

Commit:
Remove view_type from views

OCA-git-bot committed:
base_geoengine 13.0.1.0.1

Creamaster committed:
[IMP] base_geoengine: remove unnecessary hack

This hack is not needed anymore, and it cause an UI problem: the option 'geoengine' is not suggested in the selection list (backend views)

Co-authored-by: Thierry Ducrest <TDu@users.noreply.github.com>

lmarion-source committed:
changing the default SRID (correction of the previous one)

oca-travis committed:
[UPD] Update base_geoengine.pot

TDu committed:
Fix default_extent in views

Convert the extent information from string to number when opening an
empty view.
Although the default_extent is set with a default value on creation, the
field can be set to null so make sure a default value is used.

Co-authored-by: Yannick Vaucher <yannick.vaucher@camptocamp.com>

OCA-git-bot committed:
base_geoengine 13.0.1.0.2

brian10048 committed:
[LINT] pre-commit

NL66278 committed:
[FIX] base_geoengine: use shape() instead of asShape()

NL66278 committed:
[UPD] Move from travis to github actions

HekkiMelody committed:
[14.0][FIX]base_geoengine: non-admin users can see maps

yvaucher committed:
[IMP] base_geoengine: black, isort, prettier

yvaucher committed:
[MIG] base_geoengine: Migration to 14.0

yvaucher committed:
Yannick Vaucher renaming to Yannick Payot

Commit:
[UPD] Update base_geoengine.pot

OCA-git-bot committed:
[UPD] README.rst

kouffsamuel committed:
[IMP] base_geoengine: black, isort, prettier

kouffsamuel committed:
[MIG] base_geoengine : migration from odoo 14 to odoo 16

yvaucher commented:
@lmignon FYI, BTW does acsone still plan to migrate geoengine on Owl ?

yvaucher commented:
Commit to add :

https://github.com/OCA/geospatial/pull/327

kouffsamuel committed:
[ADD] base_geoengine: display field geo_edit_map with multipolygon and create multipolygon

lmignon commented:
Hi @yvaucher 

@kouffsamuel will migrate the widgets and views to owl. He is working under my supervision on this subject as part of his final thesis (supervision is perhaps not the most appropriate word, we can say that we are collaborating even if he is doing all the work and I am there to guide and advise him in what he is doing). I proposed to him to open a draft PR in order to give visibility to his work which will probably take several weeks.  He will make a second PR with the demo module so that external people can test functionally what will be done. Any comment, suggestion for evolution etc. will be welcome.  

Have a nice day.

lmignon commented on base_geoengine/static/src/js/widgets/geoengine_edit_map/field_geoengine_edit_map.xml:
```diff
@@ -1,6 +1,7 @@
 <templates xml:space="preserve">
     <t t-name="FieldGeoEngineEditMap" owl="1">
-
+        <div id="map" style="width: 100%; height: 100%;">
```
Do we need a `style` attribute on this element? You already specify a css rule on 'o_field_geo_edit_map'. In any case, if style attributes are needed, they must come from the css and not from the style attribute on the element itself.

The value for the `id` attribute must come from the widget id to avoid duplicate if the widget is used more than one time on the same view.

lmignon replied:
ping @kouffsamuel

kouffsamuel committed:
[FIX] base_geoengine: fix zip creation and refractor geo_edit_map

kouffsamuel committed:
[FIX] base_geoengine: fix retails machine and start geoengine view fix

kouffsamuel committed:
[FIX] base_geoengine: fix geoengine view renderer

kouffsamuel committed:
[ADD] base_geoengine: add left-side menu for layers

kouffsamuel committed:
[FIX] base_geoengine: display raster layers for zip and retail machines and add a menu

kouffsamuel committed:
[FIX] base_geoengine: display vector layers for zip and retail machines

dangeross commented:
Hi @kouffsamuel 

Seems like [base_geoengine/static/lib/geostats-2.0.0](https://github.com/OCA/geospatial/pull/328/commits/ce6e046c80162f3c69f439d3acb5a3632214f1ef#diff-33c2bcd08018ad8e4da200f5b91f5b34a9134a0f52b50f9fda9f8d6245e6dc46) is referencing a submodule somewhere (by mistake?)

kouffsamuel committed:
[FIX] remove ref to submodule

kouffsamuel committed:
[FIX] base_geoengine: display info box

kouffsamuel committed:
[FIX] base_geoengine: button to edit record on info box + fix display geoengine view

kouffsamuel committed:
[FIX] base_geogines: fix bugs and use opacity of the layer management

kouffsamuel committed:
[FIX] base_geoengine: code cleaning

kouffsamuel committed:
[FIX] base_geoengine: fix build

kouffsamuel committed:
[FIX] base_geoengine: fix pre-commit

kouffsamuel committed:
[IMP][FIX] base_geoengine: start model tests and fix retail machines view in geo_edit_map

kouffsamuel committed:
[FIX] base_geoengine: test-requirements

kouffsamuel committed:
[IMP] base_geoengine: end model tests

sbejaoui commented on base_geoengine/fields.py:
```diff
@@ -0,0 +1,290 @@
+        if shape_to_write.is_empty:
+            return None
+        else:
+            return shape_to_write.wkt
```
@lmignon , @kouffsamuel

I am getting an error when editing a geoField,

```sql
UPDATE "res_partner" SET "geo_point" = 'POINT (-13578158.298793 4513301.6464)';
```
```log
psycopg2.errors.InvalidParameterValue: Geometry SRID (0) does not match column SRID (3857)
```
I fixed it by setting the SRID for the given point 

 ```suggestion
            return f'SRID={self.srid};{shape_to_write.wkt}'
```

lmignon replied:
@kouffsamuel @sbejaoui This conversion was previously implemented in 10.0 by the use of the 'column_format' on the field definition https://github.com/OCA/geospatial/blob/10.0/base_geoengine/fields.py#L36 
```python
    @property
    def column_format(self):
        return "ST_GeomFromText(%%s, %s)" % (self.srid,)
```

The `convert_to_column` method seems to be the replacement of the `column_format` method.  Would it be more accurate to write?
```suggestion
            return f"ST_GeomFromText({shape_to_write.wkt}, {self.srid})"
```

lmignon commented on base_geoengine/fields.py:
```diff
@@ -0,0 +1,290 @@
+        )
+        res = cr.fetchone()
+        return cls.load_geo(res[0])
+
```
@kouffsamuel This commit is missing https://github.com/OCA/geospatial/commit/f5dfefb7d74a40c1c2684a019c7f1f609b6d8c26

kouffsamuel committed:
[IMP] base_geoengine: add tests from_lat_lonand to_lat_lon

kouffsamuel committed:
[IMP] base_geoengine: refactor geo_search

kouffsamuel committed:
[FIX] base_geoengine: current_field __leaf_to_sql

kouffsamuel committed:
[FIX] base_geoengine: Display Labels on Polygon

kouffsamuel committed:
[FIX] base_geoengine: fix srid

kouffsamuel committed:
[FIX] base_geoengine: fix column definition at install

Before this change, simple geometry column were created without inforamtion regarding the geo type nor the defined SRID. Without these information, the srid has to be expliciltely set when converting data to column format.

kouffsamuel committed:
[IMP] base_geoengine: geo_search is deprecated, normal search method had to be use. Add the possibility to have a right internal domain on the right

sbejaoui commented on base_geoengine/expressions.py:
```diff
@@ -0,0 +1,107 @@
+            query = current_operator.get_geo_intersect_sql(
+                table, left, right, params, rel_col=rel_col, rel_model=rel_model
+            )
+    return query
```
```suggestion
        case _:
            raise NotImplementedError(f'The operator {operator} is not supported')
    return query
```

kouffsamuel committed:
[FIX] base_geoengine: fix __leaf_to_sql by adding our own where_calc method with our alias to perform search method and fix tests

kouffsamuel committed:
[FIX] base_geoengine: fix fields SRID

kouffsamuel committed:
[IMP] base_geoengine : Access rules

kouffsamuel committed:
[IMP] base_geoengine: Allows to define a model on the vector layer to apply the specified model

kouffsamuel committed:
[FIX] base_geoengine: fix priority and add domain on geo_field_id

kouffsamuel committed:
[IMP] base_geoengine: this code seems not to be used

kouffsamuel committed:
[ADD] base_geoengine : add copyright

kouffsamuel committed:
[FIX] base_geoengine: Use different types of source for rasters layers

kouffsamuel committed:
[ADD] base_geoengine: add the possibility to specify a color and an opacity on geo_fields

kouffsamuel committed:
[IMP] base_geoengine: readme, refactor and documentation

kouffsamuel committed:
[FIX] base_geoengine: fix readme

kouffsamuel committed:
[IMP] base_geoengine: moving unnecessary files from the models folder

kouffsamuel committed:
[ADD] base_geoengine: order the vector layer in relation to the sequence

kouffsamuel committed:
[FIX] base_geoengine: fix tests

lmignon commented on base_geoengine/models/base.py:
```diff
@@ -0,0 +1,169 @@
+        # and do a search on standard attributes
+        # Limit and offset are managed after, we may loose a lot of performance
+        # here
+        _logger.debug(
+            "geo_search is deprecated: uses search method defined on base model"
+        )
```
@kouffsamuel We can use here the standard way to declare deprecation into python....
 In your imports add: `import warnings`

```suggestion
        warnings.warn(
           "geo_search is deprecated: uses search method defined on base model",
        DeprecationWarning)
```

lmignon reviewed (approved):
Great work @kouffsamuel ! LGTM

drewes commented:
Fantastic work. Well done.

baimont commented:
A general comment, which is quite subjective:

![image](https://user-images.githubusercontent.com/32102436/236195604-95981628-1f20-4e91-baaf-faf46cc19983.png)

You commit a lot. This makes it quite difficult to read the great categories of changes that you have done during the migration. But as I said, this is quite subjective for I myself do not commit enough and tend to make one ugle big commit for a migration, even if it's that big. Nevertheless, maybe some of these commits could be merged together?

Anyway, fantastic work! Well done!

baimont commented on base_geoengine/readme/USAGE.rst:
```diff
@@ -0,0 +1,43 @@
+--------------
+1. As a user/admin, when I am in the Geoengine Demo module and I go to the ZIP menu.
+   When I click on an item in the list view, I get to the form view showing me the different
+   information about the ZIP. We can see its ZIP, city, priority, total sales and his spatial
```
```suggestion
   information about the ZIP. We can see its ZIP, city, priority, total sales and its spatial
```

kouffsamuel commented:
> A general comment, which is quite subjective:
> 
> ![image](https://user-images.githubusercontent.com/32102436/236195604-95981628-1f20-4e91-baaf-faf46cc19983.png)
> 
> You commit a lot. This makes it quite difficult to read the great categories of changes that you have done during the migration. But as I said, this is quite subjective for I myself do not commit enough and tend to make one ugle big commit for a migration, even if it's that big. Nevertheless, maybe some of these commits could be merged together?
> 
> Anyway, fantastic work! Well done!

Thanks for your feedback.

yvaucher commented:
OL 7 :yum: 

Fantastic work.

yvaucher reviewed (commented):
On the UI, the GeoEngine Data tab on ir.ui.view form has some fields (projection and restricted map extent `restricted_extent`) that are not well displayed.

~~Do you also plan to migrate base_geoengine_demo?~~ (https://github.com/OCA/geospatial/pull/329)

yvaucher commented:
/ocabot migration base_geoengine

yvaucher commented:
@kouffsamuel If I understand correctly,  test_base_geoengine is integrated in `base_geoengine` as I see some new test files?

Or should we still migrate it?

CF: https://github.com/OCA/geospatial/blob/13.0/test_base_geoengine/tests/test_base_geoengine.py

kouffsamuel commented:
> @kouffsamuel If I understand correctly, test_base_geoengine is integrated in `base_geoengine` as I see some new test files?
> 
> Or should we still migrate it?
> 
> CF: https://github.com/OCA/geospatial/blob/13.0/test_base_geoengine/tests/test_base_geoengine.py

Yes, I do some tests in base_geoengine but I didn't take into account these tests. So it should probably still be migrated.

lmignon commented:
> > @kouffsamuel If I understand correctly, test_base_geoengine is integrated in `base_geoengine` as I see some new test files?
> > Or should we still migrate it?
> > CF: https://github.com/OCA/geospatial/blob/13.0/test_base_geoengine/tests/test_base_geoengine.py
> 
> Yes, I do some tests in base_geoengine but I didn't take into account these tests. So it should probably still be migrated.

IMO, the test_base_geoengine addon is now useless. Tests in base_geoengine covers more cases.

lmignon commented:
@yvaucher Do you agree to merge this one... I would like to move forward with all the pending PR's. If we encouter bugs we'll fix it since these addons are used in our current project.

bealdav commented:
Hi @yvaucher could you actualize your review, please. Thanks

lmignon commented:
merged with #334
